_drafts/practical_zephyr_05_west.md

```diff
@@ -0,0 +1,1246 @@
+    ├── boards
+    │   └── nrf52840dk_nrf52840.overlay
+    ├── src
+    │   └── main.c
+    ├── CMakeLists.txt
+    ├── prj.conf
+    └── west.yml  # manifest file
+```
+
+With this, we can initialize our workspace as follows:
+
+```bash
+$ west init --local app
+=== Initializing from existing manifest repository app
+--- Creating /path/to/workspace/.west and local configuration file
+=== Initialized. Now run "west update" inside /path/to/workspace.
+```
+
+Before running `west update`, let's add Zephyr as a project in `manifest.projects` just like what we've done for the [Doxygen-Awesome](https://github.com/jothepro/doxygen-awesome-css) dependency before:
+
+`app/west.yml`
+
+```yaml
+manifest:
+  version: 0.8
+
+  self:
+    path: app
+
+  projects:
+    - name: zephyr
+      revision: v3.4.0
+      url: https://github.com/zephyrproject-rtos/zephyr
+      path: deps/zephyr
+```
+
+Running `west update` clones the Zephyr repository into `deps/zephyr` and checks out the commit with the tag `v3.4.0`:
+
+```bash
+$ west update
+=== updating zephyr (deps/zephyr):
+--- zephyr: initializing
+Initialized empty Git repository in /path/to/workspace/deps/zephyr/.git/
+--- zephyr: fetching, need revision v3.4.0
+...
+HEAD is now at 356c8cbe63 release: Zephyr 3.4.0 release
+```
+
+Now, we have a complete "T2 star topology" workspace, where the Zephyr dependency is placed in a separate `deps` folder in the _topdir_ (again following the convention presented by Mike Szczys in his talk ["Manifests: Project Sanity in the Ever-Changing Zephyr World"](https://www.youtube.com/watch?v=PVhu5rg_SGY)):
+
+
+```bash
+tree --dirsfirst -a -L 3
+.  # topdir
+├── .west
+│   └── config
+├── app  # manifest repository
+│   ├── .git
+│   ├── boards
+│   │   └── nrf52840dk_nrf52840.overlay
+│   ├── src
+│   │   └── main.c
+│   ├── CMakeLists.txt
+│   ├── prj.conf
+│   └── west.yml  # manifest file
+└── deps
+    └── zephyr
+```
+
+Since we've used this application already in the previous article, we're pretty sure that it should build. However, building still fails since we didn't tell _West_ about Zephyr's extension commands yet:
+
+```bash
+$ west build --board nrf52840dk_nrf52840 app
+usage: west [-h] [-z ZEPHYR_BASE] [-v] [-V] <command> ...
+west: unknown command "build"; workspace /path/to/workspace does not define this extension command -- try "west help"
+```
+
+We can fix this by adding the `west-commands` key to the `zephyr` project: Zephyr's _West_ extensions are provided by the file `scripts/west-commands.yml` in Zephyr's repository. Using the key `west-commands`, we can provide a relative path to _West_ extension commands within the project:
+
+`app/west.yml`
+
+```yaml
+manifest:
+  version: 0.8
+
+  self:
+    path: app
+
+  projects:
+    - name: zephyr
+      revision: v3.4.0
+      url: https://github.com/zephyrproject-rtos/zephyr
+      path: deps/zephyr
+      # add Zephyr-specific West extensions
+      west-commands: scripts/west-commands.yml
```

Comment:
I might recommend skipping this for imports

Reply:
You mean you'd not add `west-commands`? Without `import` at this point it'd be necessary to keep it since otherwise the `west` extension commands would not work, but I'll add a note.

Reply:
Oh right, good point. I think leaving things here is fine since we have the later comment